pkg/package-server/server/server.go

```diff
@@ -190,7 +190,10 @@ func (o *PackageServerOptions) Run(stopCh <-chan struct{}) error {
 		return err
 	}
 
-	go sourceProvider.Run(stopCh)
+	ready := make(chan struct{})
```

Comment:
I almost want to suggest a generic `blockUntilReady` decorator, I just think it might be a little wordy in `go`.

```go
type ReadyStarter func(chan struct{})

func startAndBlockUntilReady(signaller ReadyStarter) {
  ready := make(chan struct{})
  defer close(ready)
  go signaller(ready)
  <-ready
}

func main() {
   stopCh := ...
   sourceProviderStarter := func(ready chan struct{}) {
     sourceProvider.Run(stopCh, ready)
   }
   startAndBlockUntilReady(sourceProviderStarter)
}

```
meh...might be nice if go had nice partial application, but...